src/problem_bank_scripts/problem_bank_scripts.py

```diff
@@ -80,6 +80,11 @@ def defdict_to_dict(defdict, finaldict):
                     v[k2] = dict(sorted(v2.items(), key=lambda i: i[0]))
                 
             finaldict[k] = v
+        elif hasattr(v, "dtype"):
+            try:
+                finaldict[k] = v.item()
+            except:
```

Comment:
```suggestion
            except ConstructorError:
```

This should be a ConstructorError right ([ref](https://github.com/open-resources/stats_bank/actions/runs/10519721270/job/29147553222#step:6:35))? Generally not good practice to catch all errors

Reply:
> This should be a ConstructorError right ([ref](https://github.com/open-resources/stats_bank/actions/runs/10519721270/job/29147553222#step:6:35))? Generally not good practice to catch all errors

pyyaml's `ConstructorError` would be the wrong error to use here.

A middle ground is to catch an `Exception`, such that we just don't convert if the `.item()` vall fails - I'd rather to bail out over failing catastrophically when converting the defaultdict to a standard dict if a non numpy object that has a `dtype` attribute gets passed to a data dictionary - I don't know where that could happen, but its possible.

```suggestion
            except Exception:
```